WalletWasabi.Fluent/ViewModels/Wallets/LoadingViewModel.cs

```diff
@@ -94,10 +95,11 @@ private async Task LoadWalletAsync(bool isBackendAvailable)
 			}
 
 			_isLoading = true;
+			int nodeCount = Services.HostedServices.Get<P2pNetwork>().Nodes.ConnectedNodes.Count;
 
 			await SetInitValuesAsync(isBackendAvailable).ConfigureAwait(false);
 
-			while (isBackendAvailable && RemainingFiltersToDownload > 0)
+			while (isBackendAvailable && RemainingFiltersToDownload > 0 && nodeCount > 0)
```

Comment:
@soosr Thanks for reviewing the pull request

> 1. The number of nodes is stored only once into the nodeCount variable. In the case when the LoadWalletAsync is called when the number of nodes is zero, then we have a forever loop and the user will never leave the loading screen. (In the case when the backend is available.)

Interesting. Will have to test this and figure out how to avoid it.

> 2. Even if the number of nodes is checked continuously, what if there is a connection problem and the count will be 0 forever? There should be a fallback mechanism for this as well.

Fallback mechanism makes sense.

> 3. I am not sure, but what is the number of nodes when the client is connected to its own node?

It can be zero sometimes: https://github.com/zkSNACKs/WalletWasabi/issues/6771#issuecomment-981723170

> 4. In the case when the client needs to synchronize (download and process some filters) but there is no relevant transaction on the blockchain so the client doesn't need any node to be connected because it doesn't have to download a block at all then the client will be blocked on the loading screen again for no reason.

Maybe this should be moved out of while loop and use it with some if else statement only in some cases. Will update if I could figure out a better implementation.